source/app/controllers/sessions_controller.rb

```diff
@@ -1,12 +1,39 @@
 class SessionsController < ApplicationController
 
+  def new
+    @user = User.new
+  end
+
   def index
   end
 
   def create
+    @user = User.find_by_email(params[:email])
+    if @user.nil?
+      flash[:login_error] = 'Login Error occured'
+      redirect_to '/' and return
+    end
+
+    if @user && @user.authenticate(params[:password])
```

Comment:
You already checked if `@user.nil?` and it can't be `false`, so I'd say just call `authenticate`.